versioned_docs/version-latest/protocol/security.md

```diff
@@ -1,86 +1,266 @@
+
+SSH Connection Establishment:
+
+    When a client initiates a TCP connection, the server responds with supported protocol versions and its public host key.
+    Both parties negotiate a session key using the Diffie-Hellman algorithm to ensure secure communication. This session key encrypts the entire session.
+
+User Authentication:
+
+    After establishing session encryption, user authentication begins.
+    Password authentication involves securely transmitting the user's account password.
+    SSH key pairs, consisting of public and private keys, are a recommended alternative for authentication. The client sends a key pair ID to the server, which checks for a matching public key.
+    The server encrypts a random number with the public key, sent to the client, who decrypts it with the private key.
+    The client computes an MD5 hash using the decrypted number and the shared session key, sending it back to the server for verification.
+
+In RSA, encryption and decryption use different keys: the encryption key is public, and the decryption key is private. A user creates a public key based on two large prime numbers and an auxiliary value, while keeping the prime numbers secret. Messages can be encrypted with the public key but can only be decrypted by someone who knows the prime numbers.
+
+Together, SSH and RSA provide secure, authenticated, and encrypted connections to protect sensitive information. For added security, you can create an SSH key with a passphrase.
+
+
+### Creating RSA key pair
+#### On the server, run: 
+`vi /etc/ssh/sshd_config`
+`--> PubkeyAuthentication yes`
+
+`systemctl restart ssh`
+
+#### Create rsa keys on home PC (if you don't have it yet).
+```console
+ssh-keygen
+```
+*you will get two keys:*
 
-### Use strong passwords
+                        `$HOME/.ssh/id_rsa`     Your private RSA key to keep on local PC
 
-:::caution MOST ACCOUNT HACKS HAPPEN BECAUSE OF WEAK OR STOLEN PASSWORDS 
+                        `$HOME/.ssh/id_rsa.pub` Your public RSA key to send to a server
 
-Make sure to make your password long and mix letters, numbers, and special symbols.
+#### Transfer rsa keys to the server:
+
+On a home PC: 
+
+`sudo ssh-copy-id -p 12345 user_name@server-ip-addres`
+
+This utility specifically designed for copying SSH keys to a remote server. 
+It automatically handles the key placement and permissions on the remote server, making it more convenient. 
+Use `-p` flag to specify not standard port, if you have changed it.
+
+if you have created keys before, don't overwrite it! As you will not be able to authenticate using the previous key anymore. But you can keep them somewhere else and generate next.
```

Comment:
```suggestion
If you have created keys before, don't overwrite it! As you will not be able to authenticate using the previous key anymore. But you can keep them somewhere else and generate them next.
```

Reply:
Thank you. I have improved my English with the help of your comments!